cmd/detectExecuteScan.go

```diff
@@ -17,28 +20,42 @@ func detectExecuteScan(config detectExecuteScanOptions, telemetryData *telemetry
 	// reroute command output to logging framework
 	c.Stdout(log.Writer())
 	c.Stderr(log.Writer())
-	runDetect(config, &c)
-}
 
-func runDetect(config detectExecuteScanOptions, command command.ShellRunner) {
-	// detect execution details, see https://synopsys.atlassian.net/wiki/spaces/INTDOCS/pages/88440888/Sample+Synopsys+Detect+Scan+Configuration+Scenarios+for+Black+Duck
+	fileUtils := piperutils.Files{}
+	httpClient := piperhttp.Client{}
 
-	args := []string{"bash <(curl -s https://detect.synopsys.com/detect.sh)"}
-	args = addDetectArgs(args, config)
-	script := strings.Join(args, " ")
+	err := runDetect(config, &c, &fileUtils, &httpClient)
 
-	command.SetDir(".")
-	command.SetEnv([]string{"BLACKDUCK_SKIP_PHONE_HOME=true"})
-
-	err := command.RunShell("/bin/bash", script)
 	if err != nil {
 		log.Entry().
 			WithError(err).
 			Fatal("failed to execute detect scan")
 	}
 }
 
-func addDetectArgs(args []string, config detectExecuteScanOptions) []string {
+func runDetect(config detectExecuteScanOptions, command command.ShellRunner, fileUtils piperutils.FileUtils, httpClient piperhttp.Downloader) error {
+	// detect execution details, see https://synopsys.atlassian.net/wiki/spaces/INTDOCS/pages/88440888/Sample+Synopsys+Detect+Scan+Configuration+Scenarios+for+Black+Duck
+	httpClient.DownloadFile("https://detect.synopsys.com/detect.sh", "detect.sh", nil, nil)
```

Comment:
smelly pattern from a security perspective. I understand that we don't want to vendor this script in, but at least the user should be aware that an arbitrary script is run in his name when running this script.